lib/websocket_proxy.rb

```diff
@@ -10,7 +10,12 @@ def initialize(env, console, logger)
     secure = Rack::Request.new(env).ssl?
     scheme = secure ? 'wss:' : 'ws:'
     @url = scheme + '//' + env['HTTP_HOST'] + env['REQUEST_URI']
-    @driver = WebSocket::Driver.rack(self, :protocols => %w(binary))
+
+    # VMware vCloud WebMKS SDK (console access) grabs 'binary' if offered, but then fails because in fact it's not
+    # able to use it :) We workaround this by only forcing the 'uint8utf8' protocol which actually works.
+    force_uint8 = console.protocol.to_s.end_with?('uint8utf8')
+    protocols = force_uint8 ? ['uint8utf8'] : ['binary']
+    @driver = WebSocket::Driver.rack(self, :protocols => protocols)
```

Comment:
I see no point of having the condition, is it OK for the client to always send both protocols?

Reply:
Oh, then just let's increase the readability:
```ruby
protocol = console.protocol.to_s.end_with?('uint8utf8') ? 'uint8utf8' : 'binary'
@driver = WebSocket::Driver.rack(self, :protocols => [protocol])
```